src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -3776,6 +4272,28 @@ private static string MatchCharacterClass(bool hasTextInfo, RegexOptions options
                 $"((ch = {chExpr}) < 128 ? ({Literal(bitVectorString)}[ch >> 4] & (1 << (ch & 0xF))) != 0 : global::System.Text.RegularExpressions.RegexRunner.CharInClass((char)ch, {Literal(charClass)}))";
         }
 
+        /// <summary>
+        /// Replaces <see cref="AdditionalDeclarationsPlaceholder"/> in <paramref name="writer"/> with
+        /// all of the variable declarations in <paramref name="declarations"/>.
+        /// </summary>
+        private static void ReplaceAdditionalDeclarations(HashSet<string> declarations, IndentedTextWriter writer)
+        {
+            StringBuilder sb = ((StringWriter)writer.InnerWriter).GetStringBuilder();
+            string replacement = "";
+
+            if (declarations.Count != 0)
+            {
+                var tmp = new StringBuilder().AppendLine();
+                foreach (string decl in declarations)
+                {
+                    tmp.Append(' ', writer.Indent * 4).AppendLine(decl);
```

Comment:
NIT: I suppose there is no API to automatically get the indentation on that line so that we don't have to guess here the right indentation? I'm fine with the way it is too, but I'm just thinking about what happens if we later decide to add an extra block that surrounds the declarations or something that will change the indentation of those lines, or also if we want to reuse this method somewhere else.

Reply:
Right.  But I can just grab the indent at the initial point and use that to set this later.